Gapit twilio sms and voice

sbgap opened the pull request.

sbgap committed:
feat: integrate twilio rules
- update setup.py, setting.py and views/__init__.py
- update database code with twilio rules in base.py and shema.sql
- update Alert with get_twilio_rules in alert.py
- update api accesses with twilio_rules accesses in enums.py
- add twiliorule.py in models with TwilioRule model
- add twiliorule.py in plugins with TwilioRulesHandler
- add twiliorules.py in views with api handling

sbgap committed:
fix: fix problem with finding active twilio rules

sbgap committed:
feat: add method to change sms/call text in api

- change how the sms/call text is generated in plugins/twiliorule.py
- change call method to also send sms in plugins/twilio_rule
.py

sbgap committed:
Merge remote-tracking branch 'upstream/master' into gapit_twilio_sms_and_voice

ZuperZee commented on alerta/database/backends/mongodb/base.py:
```diff
@@ -1206,6 +1206,108 @@ def delete_blackout(self, id):
+        if twilio_rule.start_time:
+            data['startTime'] = twilio_rule.start_time.hour + twilio_rule.start_time.minute / 100
+            data['endTime'] = twilio_rule.end_time.hour + twilio_rule.end_time.minute / 100
+        # if twilio_rule.days:
+        #     data['days'] = twilio_rule.days
+        # if twilio_rule.severity:
+        #     data['severity'] = twilio_rule.severity
```
Shouldn't this have been removed?

ZuperZee commented on alerta/database/backends/mongodb/base.py:
```diff
@@ -1206,6 +1206,108 @@ def delete_blackout(self, id):
+
+    def delete_twilio_rule(self, id):
+        response = self.get_db().twilio_rules.delete_one({'_id': id})
+        return True if response.deleted_count == 1 else False
```
This colud've been written much simplier
```py
 return response.deleted_count == 1
 ```

ZuperZee commented on alerta/models/alert.py:
```diff
@@ -416,6 +420,12 @@ def is_suppressed(self) -> bool:
+    def get_twilio_rules(self) -> 'list[TwilioRule]':
+        if not self.duplicate_count or self.duplicate_count == 0:
+            return [TwilioRule.from_db(twilio_rule) for twilio_rule in db.get_twilio_rules_active(self)]
+        else:
```
Unnecessary else.

sbgap committed:
fix: remove unused code
- removed commented code from database/mongodb/base.py
- removed unused exception class TwilioAuth from exceptions.py
